### PR TITLE
Update sandbox component 'opennlp-wsd' to be compatible with latest opennlp-tools release

### PR DESCRIPTION
**Pull request by mawiesne**
- adjusts opennlp-tools to 2.1.0
- adjusts parent project (org.apache.apache) to version 18
- adjusts Java language level to 11
- adds missing test resources to check whether the existing work; some do, some don't...
- ignores tests that aren't functional even with corresponding test resources

### CODE REVIEW

File: opennlp-wsd/src/main/java/opennlp/tools/cmdline/disambiguator/DisambiguatorTool.java [hunk 1]
```diff
@@ -117,11 +120,12 @@ static ObjectStream<WSDSample> openSampleData(String sampleDataName,
...
 
-    FileInputStream sampleDataIn = CmdLineUtil.openInFile(sampleDataFile);
-
-    ObjectStream<String> lineStream = new PlainTextByLineStream(
-        sampleDataIn.getChannel(), encoding);
-
-    return new WSDSampleStream(lineStream);
+    try {
+      MarkableFileInputStreamFactory factory = new MarkableFileInputStreamFactory(sampleDataFile);
+      ObjectStream<String> lineStream = new ParagraphStream(new PlainTextByLineStream(factory, encoding));
```
Comment (kinow) [inline opennlp-wsd/src/main/java/opennlp/tools/cmdline/disambiguator/DisambiguatorTool.java:125]: Sorry, not on my laptop with an IDE to confirm, but if these two are Closeable's perhaps they could be moved to the `try-with-resources`, if that makes sense.

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/LeskEvaluatorTest.java [hunk 2]
```diff
@@ -23,23 +23,24 @@
...
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class LeskEvaluatorTest {
 
   static SensevalReader seReader = new SensevalReader();
 
   @Test
-  public static void main(String[] args) {
+  @Ignore // TODO Investigate why test fails while parsing 'EnglishLS.train'
```
Comment (kinow) [inline opennlp-wsd/src/test/java/opennlp/tools/disambiguator/LeskEvaluatorTest.java:34]: :+1: maybe create an issue for later if we are not planning to work on this soon.

**Review by kinow — APPROVED** [review]
Random comments, but no blockers :+1: thanks!

**Comment by mawiesne** [comment]
@kinow I'd like to point your interest to the test resources directory and the files I added herein. What's your opinion on the bunch of data. IMHO, it is required to have some sort of tests running...; on the other hand it's quite a mass. Any ideas how we can achieve a tradeoff here, that is, have it more lightweight?

**Comment by kinow** [comment]
> @kinow I'd like to point your interest to the test resources directory and the files I added herein. What's your opinion on the bunch of data. IMHO, it is required to have some sort of tests running...; on the other hand it's quite a mass. Any ideas how we can achieve a tradeoff here, that is, have it more lightweight?

Thanks for pointing that out, @mawiesne . I am using the GH UI, and skipped the test files without noticing their sizes.

Does compressing these files help us here? If so we could include them as gzip/some other compression algo, and decompress when running the tests?

Otherwise, what I used in another project that needed some large files for tests was the following setup:

- enabled LFS in git
- create tests using the large files in a separate suite
- disable the suite by default, with a toggle in the build to enable that (i.e. you assume the user/dev will have enabled LFS and pulled the large files)

The final alternative I can think of is storing the files in some ASF host somewhere, if possible, and do something similar to the setup with Git LFS, but asking users to download these files to run the tests in certain suites.

Not sure if really helpful.

**Comment by mawiesne** [comment]
> create tests using the large files in a separate suite

Could be covered via a MVN profile in a separate PR.

For now, it would be okay to add those larger (binary) files to an existing sandbox project, right? Making those structures prettier would be a next step, oc.

**Comment by kinow** [comment]
>For now, it would be okay to add those larger (binary) files to an existing sandbox project, right? Making those structures prettier would be a next step, oc.

How large would they be? I am not syncing the project every day, so it won't affect me. As long as it's in 10's of MB's that should be fine, I think. In Apache Commons Imaging I think we included some 4, 5MB images needed for testing at one point, but we are trying to keep the test data small (and the tests there don't need the same amount as for a model in opennlp).

File: opennlp-wsd/src/main/java/opennlp/tools/cmdline/disambiguator/DisambiguatorTool.java (hunk 1 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/cmdline/disambiguator/DisambiguatorTool.java]: Fixed.

**Comment by mawiesne** [comment]
> How large would they be?

- senseval3: ~6.9 MB (uncompressed)
- opennlp models: ~8.9 MB (compressed/binary)

**Comment by mawiesne** [comment]
> How large would they be?

- senseval3: ~6.9 MB (uncompressed)
- opennlp models: ~8.9 MB (compressed/binary)

**Comment by kinow** [comment]
> > How large would they be?
> 
>     * senseval3: ~6.9 MB (uncompressed)
> 
>     * opennlp models: ~8.9 MB (compressed/binary)

I think these are fine for now, especially given we can try to reduce the size or use another approach in a follow-up issue :+1:

**Comment by kinow** [comment]
>I gziped some plain resource and implemented code to handle reading in compressed forms. This way, we are down to:

Excellent! I think these sizes are good for now, so no need to worry about that for a while :+1: Thanks @mawiesne !

File: opennlp-wsd/pom.xml [hunk 3]
```diff
@@ -25,20 +25,20 @@
 	<parent>
 		<groupId>org.apache</groupId>
 		<artifactId>apache</artifactId>
-		<version>13</version>
+		<version>18</version>
```
Comment (rzo1) [inline opennlp-wsd/pom.xml:28]: Can we go for `29` ? We can update it in `opennlp` as well (mostly build support, etc.)

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java [hunk 4]
```diff
@@ -68,11 +72,11 @@ public class WSDHelper {
       "RBR", "RBS", "VB", "VBD", "VBG", "VBN", "VBP", "VBZ" };
 
   // List of Negation Words
-  public static ArrayList<String> negationWords = new ArrayList<String>(
+  public static ArrayList<String> negationWords = new ArrayList<>(
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:75]: There is a switch from `HashMap` to `Map` -> also switch from `ArrayList` to `List` ?

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java [hunk 5]
```diff
@@ -480,14 +482,10 @@ public static HashMap<String, Object> getEnglishWords(String dict) {
...
 
-    ArrayList<String> adjective = new ArrayList<String>(Arrays.asList("JJ",
-        "JJR", "JJS"));
-    ArrayList<String> adverb = new ArrayList<String>(Arrays.asList("RB", "RBR",
-        "RBS"));
-    ArrayList<String> noun = new ArrayList<String>(Arrays.asList("NN", "NNS",
-        "NNP", "NNPS"));
-    ArrayList<String> verb = new ArrayList<String>(Arrays.asList("VB", "VBD",
-        "VBG", "VBN", "VBP", "VBZ"));
+    List<String> adjective = new ArrayList<>(Arrays.asList("JJ", "JJR", "JJS"));
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:485]: The `new ArrayList()` calls are not needed.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java [hunk 6]
```diff
@@ -606,7 +604,7 @@ public static boolean areStringArraysEqual(String[] array1, String[] array2) {
 
   public static ArrayList<WordPOS> getAllRelevantWords(String[] sentence) {
 
-    ArrayList<WordPOS> relevantWords = new ArrayList<WordPOS>();
+    ArrayList<WordPOS> relevantWords = new ArrayList<>();
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:607]: List?

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java [hunk 7]
```diff
@@ -631,7 +629,7 @@ public static ArrayList<WordPOS> getAllRelevantWords(String[] sentence) {
   public static ArrayList<String> StemWordWithWordNet(WordPOS wordToStem) {
     if (wordToStem == null)
       return null;
-    ArrayList<String> stems = new ArrayList<String>();
+    ArrayList<String> stems = new ArrayList<>();
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:632]: List?

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/datareader/SemcorReaderExtended.java [hunk 8]
```diff
@@ -200,7 +202,7 @@ private ArrayList<Sentence> readFile(String file) {
    */
   private ArrayList<WSDSample> getSemcorOneFileData(String file, String wordTag) {
 
-    ArrayList<WSDSample> setInstances = new ArrayList<WSDSample>();
+    ArrayList<WSDSample> setInstances = new ArrayList<>();
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/datareader/SemcorReaderExtended.java:205]: List?

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/datareader/SensevalReader.java [hunk 9]
```diff
@@ -218,9 +240,9 @@ public ArrayList<WSDSample> getSensevalData(String wordTag) {
                       String textAfter = nChild.getChildNodes().item(2)
                           .getTextContent();
 
-                      ArrayList<String> textBeforeTokenzed = new ArrayList<String>(
+                      ArrayList<String> textBeforeTokenzed = new ArrayList<>(
```
Comment (rzo1) [inline opennlp-wsd/src/main/java/opennlp/tools/disambiguator/datareader/SensevalReader.java:243]: No need for `new` Operator, if we use `List` interface

**Review by rzo1 — APPROVED** [review]
Added some (non blocking) comments.

File: opennlp-wsd/pom.xml (hunk 3 above)
Comment (mawiesne) [reply opennlp-wsd/pom.xml:28]: Good idea, I wanted to keep it in sync with opennlp core project. Will take note for next PRs, once the core is referencing "29". Feel free to open up an issue there, so make this happen soon.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java (hunk 4 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:75]: Unclear to me.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java (hunk 6 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:607]: Will check.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java (hunk 4 above)
Comment (rzo1) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:75]: List?

File: opennlp-wsd/pom.xml (hunk 3 above)
Comment (rzo1) [reply opennlp-wsd/pom.xml:28]: https://issues.apache.org/jira/browse/OPENNLP-1452

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java (hunk 6 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:607]: Method signature returns ArrayList and is used quite frequently under this assumption. Won't change (in this PR).

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java (hunk 7 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/WSDHelper.java:632]: see other comment, not this PR.

File: opennlp-wsd/src/main/java/opennlp/tools/disambiguator/datareader/SemcorReaderExtended.java (hunk 8 above)
Comment (mawiesne) [reply opennlp-wsd/src/main/java/opennlp/tools/disambiguator/datareader/SemcorReaderExtended.java:205]: see other comment, not this PR.

**Comment by mawiesne** [comment]
> Added some (non blocking) comments.

@rzo1 Comments resolved where applicable.

**Review by rzo1 — APPROVED** [review]
Thanks @mawiesne - LGTM

**Pull request merged**